### PR TITLE
CTCTRADERS-660: Check mode navigation 

### PR DESCRIPTION
**Pull request by ashwinirudrappa**

### CODE REVIEW

File: app/navigation/Navigator.scala [hunk 1]
```diff
@@ -60,10 +61,10 @@ class Navigator @Inject()() {
   }
 
   private def isDeclarationForSomeoneElse(ua: UserAnswers, mode: Mode): Call = {
-    if(ua.get(DeclarationForSomeoneElsePage).contains(true)) {
-      routes.RepresentativeNameController.onPageLoad(ua.id, mode)
-    } else {
-      routes.MovementDetailsCheckYourAnswersController.onPageLoad(ua.id)
+    (ua.get(DeclarationForSomeoneElsePage), ua.get(RepresentativeNamePage), mode) match {
+      case (Some(true), None, CheckMode) => routes.RepresentativeNameController.onPageLoad(ua.id, NormalMode)
```
Comment (garethgooch) [inline app/navigation/Navigator.scala:65]: We appear to be changing mode here which doesn't seem correct?

File: conf/views/declarationPlace.njk [hunk 2]
```diff
@@ -50,7 +50,8 @@
           }) }}
 
           {{ govukButton({
-            text: messages("site.continue")
+            text: messages("site.continue"),
+            attributes: {id:"submit"}
```
Comment (garethgooch) [inline conf/views/declarationPlace.njk:54]: Are we missing this on all generated pages?

File: app/navigation/Navigator.scala (hunk 1 above)
Comment (ashwinirudrappa) [reply app/navigation/Navigator.scala:65]: @garethgooch Its the way we followed in arrival as well, first user selects No on "declaration for someone else" question he reached CYA page there he changes from No to Yes then it should flow the way it normally it flows in Normal mode

File: conf/views/declarationPlace.njk (hunk 2 above)
Comment (ashwinirudrappa) [reply conf/views/declarationPlace.njk:54]: @garethgooch Will check

**Pull request merged**